### DISCUSSION TITLE
Valetudo 2021.10.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1119] [Release Discussions]
<div align="center">
    <img src="https://github.com/Hypfer/Valetudo/blob/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2021.10.0</h2></p>
</div>

Another release with a lot of UI changes by @ccoors. Also, there's an Android companion App now.

## Android Companion App

Thanks to the help of @TheLastProject who had built the first prototype in less than 18h after initially mentioning the idea on the telegram group, there's now an Android Companion App for Valetudo.

Don't worry, it is completely optional. All it does and aims to do is to find Valetudo instances on your network via Bonjour and help you with the provisioning (configuring Wi-Fi) process of new robots with Valetudo installed.
This can be helpful for example if you were to give your non-linux-skilled parents a rooted robot for Christmas or whatever.

It's available on F-Droid and Google Play and of course open source.

For more information, please check out [the docs](https://valetudo.cloud/pages/companion_apps/valetudo_companion.html).

[<img src="https://github.com/Hypfer/valetudo-companion/raw/master/fastlane/metadata/android/en-US/images/phoneScreenshots/screenshot-02.png" width=250>](https://github.com/Hypfer/valetudo-companion/raw/master/fastlane/metadata/android/en-US/images/phoneScreenshots/screenshot-02.png)[<img src="https://github.com/Hypfer/valetudo-companion/raw/master/fastlane/metadata/android/en-US/images/phoneScreenshots/screenshot-03.png" width=250>](https://github.com/Hypfer/valetudo-companion/raw/master/fastlane/metadata/android/en-US/images/phoneScreenshots/screenshot-03.png)[<img src="https://github.com/Hypfer/valetudo-companion/raw/master/fastlane/metadata/android/en-US/images/phoneScreenshots/screenshot-05.png" width=250>](https://github.com/Hypfer/valetudo-companion/raw/master/fastlane/metadata/android/en-US/images/phoneScreenshots/screenshot-05.png)

## UI Changes

@ccoors provided very nice previews displaying their changes in each PR, which is why I will just copy-paste them here for your convenience.
Also, I'm lazy

<img src="https://user-images.githubusercontent.com/8963459/135535188-eac35c13-e42f-40d9-95cc-d861077734f9.png" width="750" alt="wifi config and more">

<img src="https://user-images.githubusercontent.com/8963459/134895551-0694425c-352e-4a58-b4a9-486be4bf1693.png" width="300" alt="voice pack management"><img src="https://user-images.githubusercontent.com/8963459/135510056-c7478ae7-33cb-4f83-88cf-0f41e9b5e8ab.png" width="300" alt="manual control">

<img src="https://user-images.githubusercontent.com/8963459/135677738-038fb375-e9c8-444d-a8ce-3f3f032a2f43.png" width="750" alt="more settings">

## Logo Cleanup

As you might've noticed, the logo slightly changed.
This avoids the bug in some people, which made them read it as *aletudo* when looking at it.

## Valetudog

The thing from the telegram group finally got a name: Valetudog
It's now some kind of mascot I guess.

<img src="https://github.com/Hypfer/Valetudo/blob/master/assets/misc/valetudog.svg" width="300" alt="valetudog">

I also bought a lot of stickers of it

![image](https://user-images.githubusercontent.com/974410/135687512-e49f8bae-97ab-4d5c-bb8b-ba989eea1518.png)

Unfortunately, I have no idea what to do with them.
They did turn out quite nice though.

## Zoned Cleanup MQTT changes

Starting with this release, it's no longer possible to clean multiple zone presets at once via MQTT.
That should've never been possible in the first place since it's conceptually wrong as you can only call a single preset at a time. Sorry about that.

Note that this is a **breaking change** and will require updating of e.g. your Home Assistant automations.
Instead of an array, you now have to send just the preset ID as a string.

### Features

* **ui:** Consumables bottom view sensors ([#1104](https://github.com/Hypfer/Valetudo/issues/1104)) ([2f83d8e](https://github.com/Hypfer/Valetudo/commit/2f83d8ea02aaefde8f132909071ca9bf7367de74))
* **ui:** DoNotDisturb configuration & Masonry on large screens ([#1112](https://github.com/Hypfer/Valetudo/issues/1112)) ([935f903](https://github.com/Hypfer/Valetudo/commit/935f903981ad5f98e9224cae143396b353a195db))
* **ui:** HTTP Basic Auth and NTP client config ([#1118](https://github.com/Hypfer/Valetudo/issues/1118)) ([2371878](https://github.com/Hypfer/Valetudo/commit/2371878cd48e19b57fce61b4200d76a1cc30257b))
* **ui:** Manage voice packs ([#1108](https://github.com/Hypfer/Valetudo/issues/1108)) ([0d02c27](https://github.com/Hypfer/Valetudo/commit/0d02c278edaf1ea60810990455b33175b957acc7))
* **ui:** Manual control ([#1114](https://github.com/Hypfer/Valetudo/issues/1114)) ([7e9ef04](https://github.com/Hypfer/Valetudo/commit/7e9ef04b2fa5d8fb4d5f7e8d3fa88366592873ea))
* **ui:** Robot settings ([#1097](https://github.com/Hypfer/Valetudo/issues/1097)) ([4b43b37](https://github.com/Hypfer/Valetudo/commit/4b43b370663ca44f03d82ba828026169075d670c))
* **ui:** Wifi configuration ([#1116](https://github.com/Hypfer/Valetudo/issues/1116)) ([3e71184](https://github.com/Hypfer/Valetudo/commit/3e7118414b6b190750e3f606c93e1b052d78b898))
* **vendor.dreame:** Add Dreame D9 Pro ([cc50a8c](https://github.com/Hypfer/Valetudo/commit/cc50a8c6e4cab7a04c68ca61d21e59e15a3a59cb))

### Bug Fixes

* **mqtt:** Fix MQTT Zone cleanup ([fdabcc5](https://github.com/Hypfer/Valetudo/commit/fdabcc52be3553f01648997ec714133e27a6e7cc))
* **ui:** Allow start and home basic commands on error state ([5fd0354](https://github.com/Hypfer/Valetudo/commit/5fd0354eabd91aa50cc42120bdabdf6947aaf604))
* **ui:** Blue line between buttons in PendingMapChangeEventControl ([196a29e](https://github.com/Hypfer/Valetudo/commit/196a29e1549ffd46fddcd78c7c8bee9b2d1e8c21))
* **ui:** Fix mobile Icons in new frontend ([#1113](https://github.com/Hypfer/Valetudo/issues/1113)) ([313caa1](https://github.com/Hypfer/Valetudo/commit/313caa10ea0a246ea05dd4ba3fca83e374e75cdc))
* **ui:** Import ToggleButton from core ([da0cce5](https://github.com/Hypfer/Valetudo/commit/da0cce588eee9c6065bb2ad4f2890b19cd8c5e59))
* **ui:** RatioBar background color ([c00d174](https://github.com/Hypfer/Valetudo/commit/c00d1749c26bec217a60a529166c5f8bf6de3dba))
* **ui:** Remove persistent data hint ([6777f3a](https://github.com/Hypfer/Valetudo/commit/6777f3aa7bbf4ef5afea5a3ba8fdecd2aa8f150e))
* **ui:** repair light mode ([#1103](https://github.com/Hypfer/Valetudo/issues/1103)) ([842aea1](https://github.com/Hypfer/Valetudo/commit/842aea125b97a33adbf28c3b0c9d1cf84c069943))
* **ui:** Root grid height ([957463a](https://github.com/Hypfer/Valetudo/commit/957463a9de34e4e7c3fafe5d94fdd6360b4a2f88))
* **ui:** Root grid height ([1d9a244](https://github.com/Hypfer/Valetudo/commit/1d9a244412b235ed2da71d594cb77d00fd9eae45))
* **ui:** Stop events badge flashing when it reloads ([#1115](https://github.com/Hypfer/Valetudo/issues/1115)) ([007926b](https://github.com/Hypfer/Valetudo/commit/007926b90881664e9b0c718e7c3ef21ad306e233))
* **webserver:** Fix reflected xss vulnerability ([36a3bae](https://github.com/Hypfer/Valetudo/commit/36a3bae630dfc84c6a2e610ef35e93a002470857))

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2021.10.0'>Valetudo 2021.10.0</a>.</em>

**Comment by ghost** [comment]
Good job as always on this project!

Just curious as i have not been following the discussions on telegram in the last weeks.
What is the reason you consider the possibility of cleaning multiple zones with one mqtt call as bad?
I use manually defined zones for every room as i like full control of what i consider to be a room. Can segments be manually defined as well? I have not looked in to it much yet as zones work perfectly well.

**Reply by Hypfer** [reply]
The idea with zone presets was that one preset may contain 1..n actual zones with n being limited by the robots firmware.

Therefore, you'd have a preset "Kitchen & Bathroom" or similar to just clean that

**Reply by ghost** [reply]
Ok, but if i want to clean two zones with a single mqtt call from HASS, i can not do this anymore?
I need to wait until the first zone is finished cleaning and then issue zone clean command for second zone?
If that is the case this seems like a downgrade in usability.

**Reply by pluegh** [reply]
Hi, thanks a ton for the great work. I just wanted to point out that the documentation is now broken at least [here](https://valetudo.cloud/pages/integrations/mqtt.html#cleansegmentsclean) and also [here](https://valetudo.cloud/pages/integrations/home-assistant-integration.html).

How can we get the same functionality in the new release? Do the commands issued via MQTT queue up? I am sure this feature is very important to many users.

**Reply by Hypfer** [reply]
Broken in what way?

**Reply by pluegh** [reply]
Sorry, I confused *zones* and *segments*! Shame on me, but maybe this clarification is at least useful for others. Thanks!

**Reply by ccoors** [reply]
There's broken JSON in the first link

**Comment by CharlesDeCholet** [comment]
hi,
i tried to install it on a new Dreame D9 and it did not install, installing the 2021.09.01 then updating worked
thanks for your work
